packages/http-client-csharp/generator/Microsoft.Generator.CSharp.ClientModel/src/Providers/RestClientProvider.cs

```diff
@@ -341,7 +341,8 @@ private static void GetParamInfo(Dictionary<string, ParameterProvider> paramMap,
             else
             {
                 var paramProvider = paramMap[inputParam.Name];
-                if (paramProvider.Type.IsEnum)
+                type = paramProvider.Field is null ? paramProvider.Type : paramProvider.Field.Type;
```

Comment:
Why do we need to check the Field.Type? Can this be different than the paramProvider.Type?

Reply:
Yes. for apiVersion parameter, it may define ad enum(see following), but we will convert it to string type for the apiVersion Field.
e.g.
```
@versioned(Versions)
@service({
  title: "ServerPathMultiple",
})
@server(
  "{endpoint}/server/path/multiple/{apiVersion}",
  "Test server with path parameters.",
  {
    @doc("Pass in http://localhost:3000 for endpoint.")
    endpoint: url,

    @doc("Pass in v1.0 for API version.")
    apiVersion: Versions,
  }
)
namespace Server.Path.Multiple;

@doc("Service versions")
enum Versions {
  @doc("Version 1.0")
  v1_0: "v1.0",
}
```